### PR TITLE
Recharts 3.0.0 - Revert using victory-vendor 

### PR DESCRIPTION
**Pull request by nikolasrieble**
## Description
This PR: 
- revert using victory-vendor for d3

This PR should not be merged into master! It is a base branch for further required changes:
(1) Upgrade d3
(2) Reproduce and fix all known issues
(3) Test and iterate ...

## Related Issue
Resolves https://github.com/recharts/recharts/issues/3676

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,8 +105,7 @@
     "@testing-library/user-event": "^14.3.0",
     "@types/classnames": "^2.2.9",
     "@types/d3-interpolate": "^3.0.1",
-    "@types/d3-scale": "^4.0.2",
-    "@types/d3-shape": "^3.1.0",
+
```
Comment (nikolasrieble) [inline package.json:108]:
```suggestion
```

File: demo/component/LineChart.tsx [hunk 2]
```diff
@@ -584,7 +584,8 @@ export default class Demo extends Component<any, any> {
 
         <p>LineChart of discrete values</p>
         <div className="line-chart-wrapper">
-          <LineChart width={400} height={400} data={data01} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
+          <LineChart
+            width={400} height={400} data={data01} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
```
Comment (nikolasrieble) [inline demo/component/LineChart.tsx:588]:
```suggestion
          <LineChart width={400} height={400} data={data01} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
```

File: package.json [hunk 3]
```diff
@@ -101,10 +103,9 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^12.1.5",
     "@testing-library/user-event": "^14.3.0",
-    "@types/classnames": "^2.2.9",
     "@types/d3-interpolate": "^3.0.1",
     "@types/d3-scale": "^4.0.2",
-    "@types/d3-shape": "^3.1.0",
+    "@types/d3-shape": "^3.1.1",
```
Comment (nikolasrieble) [inline package.json:108]:
```suggestion
    "@types/d3-shape": "^3.1.0",
```

File: package.json [hunk 4]
```diff
@@ -101,7 +103,6 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^12.1.5",
     "@testing-library/user-event": "^14.3.0",
-    "@types/classnames": "^2.2.9",
     "@types/d3-interpolate": "^3.0.1",
```
Comment (nikolasrieble) [inline package.json:106]:
```suggestion
    "@types/classnames": "^2.2.9",
    "@types/d3-interpolate": "^3.0.1",
```

**Comment by nikolasrieble** [comment]
The only tests that fail are those using
```
import userEvent from '@testing-library/user-event';
```
and the tests only fail with node v14.21.3. Reproducible locally.

**Comment by ckifer** [comment]
Changes themselves look good in terms of the dependency replacement, etc.

**Pull request closed without merging**